src/app/drive/components/ItemDetailsDialog/ItemDetailsDialog.tsx

```diff
@@ -119,6 +123,37 @@ const ItemDetailsDialog = ({
     onClose();
   }
 
+  function getSharedLocation(item: DriveItemDetails, ancestorPathNames: string[]): string {
+    let location = translate('sideNav.shared');
+    if (item.isFolder && ancestorPathNames.length > 0) {
+      location += '/' + ancestorPathNames.shift();
+    }
+    return location;
+  }
+
+  function getRegularLocation(item: DriveItemDetails, ancestorPathNames: string[]): string {
+    if (item.isFolder) {
+      ancestorPathNames.pop();
+    }
+    return item.view + (ancestorPathNames.length > 0 ? '/' + ancestorPathNames.join('/') : '');
+  }
+
+  function getLocation(item: DriveItemDetails, ancestors: DriveItemData[]): string {
+    const itemViewName = item.view;
+    const ancestorPathNames = ancestors.map((ancestor) => getItemPlainName(ancestor)).reverse();
+    ancestorPathNames.shift(); // Remove root parent
+
+    let location = '/';
+
+    if (itemViewName === 'Shared') {
+      location += getSharedLocation(item, ancestorPathNames);
+      return location;
+    }
+
+    location += getRegularLocation(item, ancestorPathNames);
+    return location;
+  }
+
```

Comment:
Remember to add tests for these functions, you will probably need to extract them from this file to add the unit tests